pennylane/fermi/conversion.py

```diff
@@ -104,6 +111,10 @@ def _(fermi_operator: FermiWord, ps=False, wire_map=None):
                 }
             )
 
+    for pw in qubit_operator:
+        if tol and abs(qml.math.imag(qubit_operator[pw])) <= tol:
```

Comment:
is it common to have operators with imaginary components _exactly_ equal to zero? If so, it would be weird to leave them just because the user didn't provide a tolerance (also, a tolerance of 0 would be ignored by this condition). Maybe instead, can you set the default tol to 0 and always do this check? Unless you're worried about performance and want this to be an opt-in thing, then this is fine.

Reply:
The Jordan-Wigner mapping gives an operator with complex coefficients. In some cases, e.g., a molecular Hamiltonian, the imaginary part is zero. The idea is that this function should do nothing to the coefficients unless a tolerance is provided. We might provide a default value of 0 but I still like the users to make the decision.